src/main/scala/chisel3/aop/injecting/InjectingTransform.scala

```diff
@@ -0,0 +1,46 @@
+// See LICENSE for license details.
+
+package chisel3.aop.injecting
+
+import firrtl.{ChirrtlForm, CircuitForm, CircuitState, Transform, ir}
+
+import scala.collection.mutable
+
+/** Appends statements contained in [[InjectStatement]] annotations to the end of their corresponding modules
+  *
+  * Implemented with Chisel Aspects and the [[chisel3.aop.injecting]] library
+  */
+class InjectingTransform extends Transform {
+  override def inputForm: CircuitForm = ChirrtlForm
+  override def outputForm: CircuitForm = ChirrtlForm
+
+  override def execute(state: CircuitState): CircuitState = {
+
+    val addStmtMap = mutable.HashMap[String, Seq[ir.Statement]]()
+    val addModules = mutable.ArrayBuffer[ir.DefModule]()
+
+    // Populate addStmtMap and addModules, return annotations in InjectStatements, and omit InjectStatement annotation
+    val newAnnotations = state.annotations.flatMap {
+      case InjectStatement(mt, s, addedModules, annotations) =>
+        addModules ++= addedModules
+        addStmtMap(mt.module) = s +: addStmtMap.getOrElse(mt.module, Nil)
+        annotations
+      case other => Seq(other)
+    }
+
+    // Append all statements to end of corresponding modules
+    val newModules = state.circuit.modules.map { m: ir.DefModule =>
+      m match {
+        case m: ir.Module if addStmtMap.contains(m.name) =>
+          m.copy(body = ir.Block(m.body +: addStmtMap(m.name)))
+        case m: _root_.firrtl.ir.ExtModule if addStmtMap.contains(m.name) =>
+          ir.Module(m.info, m.name, m.ports, ir.Block(addStmtMap(m.name)))
```

Comment:
Should this case be allowed? This seems to allow you to override the defname of an `ExtModule` using aspects. However, this name override is being derived from an `ir.Statement`. This seems odd... Is there an intended use case for this right now?

Reply:
So the name is from the `InjectStatement` which is an annotation, and the module name inside it refers to the module it is injecting the statements to. I had decided its ok to inject statements into an `ExtModule` by turning it into a normal `Module`. Do you think that is ok?

Reply:
Ah, it's rewriting to an `ir.Module`. I missed that.

I think there should be some documentation about this, but I think this behavior makes sense. An `ExtModule` is just something with a to-be-defined implementation. That could be defined in Verilog, but it could be defined by an aspect. Nice thinking ahead here.